### DISCUSSION TITLE
Cloth and hair simulation

### DISCUSSION THREAD

**Discussion by ib00** [taichi-dev/taichi#4709] [Q&A]
Are there any examples of 
a) clothing simulation
b) hair simulation

using Taichi?

**Comment by k-ye** [comment]
cc @YuCrazing @FantasyVR

**Reply by FantasyVR** [reply]
1. There are some examples of cloth simulation in Taichi source code such as:  [mass_spring_3d_ggui.py](https://github.com/taichi-dev/taichi/blob/master/python/taichi/examples/ggui_examples/mass_spring_3d_ggui.py) , [implicit_mass_spring.py](https://github.com/taichi-dev/taichi/blob/master/python/taichi/examples/simulation/implicit_mass_spring.py).
2. A simple [spring simulation](https://github.com/FantasyVR/magicMirror/blob/main/xpbd.py) could be used to simulate hairs.
